$MirOS$
$OpenBSD: patch-Source_CMakeLists_txt,v 1.3 2013/06/25 07:16:46 dcoppa Exp $

commit 480e924daedee520d18c8a9c0b9823fef205b57a
Author: David Coppa <dcoppa@openbsd.org>
Date:   Fri Jun 21 11:13:35 2013 +0200

OpenBSD: Enable ELF parsing and editing

OpenBSD provides ELF ABI declarations in

#include <stdint.h>
#include <elf_abi.h>

Teach the platform check and cmELF implementation to use these.

--- Source/CMakeLists.txt.orig	Tue Jul  2 13:41:15 2013
+++ Source/CMakeLists.txt	Sun Aug  4 15:41:21 2013
@@ -11,7 +11,13 @@
 #=============================================================================
 include(CheckIncludeFile)
 # Check if we can build support for ELF parsing.
-CHECK_INCLUDE_FILE("elf.h" HAVE_ELF_H)
+if(CMAKE_CXX_PLATFORM_ID MATCHES "OpenBSD")
+  CHECK_INCLUDE_FILES("stdint.h;elf_abi.h" HAVE_ELF_H)
+elseif(CMAKE_CXX_PLATFORM_ID MATCHES "MirBSD")
+  CHECK_INCLUDE_FILES("stdint.h;elf_abi.h" HAVE_ELF_H)
+else()
+  CHECK_INCLUDE_FILE("elf.h" HAVE_ELF_H)
+endif()
 if(HAVE_ELF_H)
   set(CMAKE_USE_ELF_PARSER 1)
 else()
